tools/deb/istio-iptables.sh

```diff
@@ -0,0 +1,181 @@
+iptables -t nat -D OUTPUT -p tcp -j ISTIO_OUTPUT 2>/dev/null
+
+# Flush and delete the istio chains
+iptables -t nat -F ISTIO_OUTPUT 2>/dev/null
+iptables -t nat -X ISTIO_OUTPUT 2>/dev/null
+iptables -t nat -F ISTIO_INBOUND 2>/dev/null
+iptables -t nat -X ISTIO_INBOUND 2>/dev/null
+iptables -t nat -F ISTIO_REDIRECT 2>/dev/null
+iptables -t nat -X ISTIO_REDIRECT 2>/dev/null
+
+if [ "${1:-}" = "clean" ]; then
+  # Only cleanup, don't add new rules.
+  exit 0
+fi
+
+# Create a new chain for redirecting inbound traffic to the common Envoy port.
+# In the ISTIO_INBOUND and ISTIO_OUTBOUND chains, '-j RETURN' bypasses Envoy
+# and '-j ISTIO_REDIRECT' redirects to Envoy.
+iptables -t nat -N ISTIO_REDIRECT
+iptables -t nat -A ISTIO_REDIRECT -p tcp -j REDIRECT --to-port ${ENVOY_PORT:-15001}
+
+# Handling of inbound ports. Traffic will be redirected to Envoy, which will process and forward
+# to the local service. If not set, no inbound port will be intercepted by istio iptables.
+if [ -n "${ISTIO_INBOUND_PORTS:-}" ]; then
+  iptables -t nat -N ISTIO_INBOUND
+  iptables -t nat -A PREROUTING -p tcp -j ISTIO_INBOUND
+
+  # Makes sure SSH is not redirectred
+  iptables -t nat -A ISTIO_INBOUND -p tcp --dport 22 -j RETURN
+
+  if [ "${ISTIO_INBOUND_PORTS:-}" == "*" ]; then
+       for port in ${ISTIO_LOCAL_EXCLUDE_PORTS:-}; do
+          iptables -t nat -A ISTIO_INBOUND -p tcp --dport ${port} -j RETURN
+       done
+       iptables -t nat -A ISTIO_INBOUND -p tcp -j ISTIO_REDIRECT
+  else
+      for port in ${ISTIO_INBOUND_PORTS}; do
+          iptables -t nat -A ISTIO_INBOUND -p tcp --dport ${port} -j ISTIO_REDIRECT
+      done
+  fi
+fi
+
+# TODO: change the default behavior to not intercept any output - user may use http_proxy or another
+# iptables wrapper (like ufw). Current default is similar with 0.1
+
+# Create a new chain for selectively redirecting outbound packets to Envoy.
+iptables -t nat -N ISTIO_OUTPUT
+
+# Jump to the ISTIO_OUTPUT chain from OUTPUT chain for all tcp traffic.
+iptables -t nat -A OUTPUT -p tcp -j ISTIO_OUTPUT
+
+# Redirect app calls to back itself via Envoy when using the service VIP or endpoint
+# address, e.g. appN => Envoy (client) => Envoy (server) => appN.
+iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT
+
+for uid in ${ENVOY_UID}; do
+  # Avoid infinite loops. Don't redirect Envoy traffic directly back to
+  # Envoy for non-loopback traffic.
+  iptables -t nat -A ISTIO_OUTPUT -m owner --uid-owner ${uid} -j RETURN
+done
+
+# Skip redirection for Envoy-aware applications and
+# container-to-container traffic both of which explicitly use
+# localhost.
+iptables -t nat -A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN
+
+IFS=,
+if [ -n "${IP_RANGES_INCLUDE:-}" ]; then
+    for cidr in ${IP_RANGES_INCLUDE}; do
+        iptables -t nat -A ISTIO_OUTPUT -d ${cidr} -j ISTIO_REDIRECT
+    done
+    iptables -t nat -A ISTIO_OUTPUT -j RETURN
+else
+    iptables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT
+fi
+
```

Comment:
is this officially replacing prepare_proxy.sh ? is this the script thats going to run in every pod?

Reply:
This is moved from the proxy repo - it is used for raw VMs.

Eventually need to replace or merge with prepare_proxy - main difference is the default capture policy, on raw vm we don't intercecept all, just the cidrs.